sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -17,22 +56,299 @@ class TableCheckpointStore:
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+            Credentials provided here will take precedence over those in the connection string.
+        :keyword str api_version:
+            The Storage API version to use for requests. Default value is '2018-03-28'.
+        :returns: A table checkpoint store.
+        :rtype: ~azure.eventhub.extensions.checkpointstoretable.TableCheckpointStore
+        """
+        endpoint, credential = parse_connection_str(
+            conn_str=conn_str, credential=None, keyword_args=kwargs
+        )
+        return cls(endpoint, table_name=table_name, credential=credential, **kwargs)
 
-    def list_ownership(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
+    def __enter__(self):
+        self._table_client.__enter__()
+        return self
 
-    def list_checkpoints(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
+    def __exit__(self, *args):
+        self._table_client.__exit__(*args)
+
+    @classmethod
+    def _create_ownership_entity(cls, ownership):
+        """
+        Create a dictionary with the `ownership` attributes.
+        """
+        ownership_entity = {
+            "PartitionKey": "{} {} {} Ownership".format(
+                ownership["fully_qualified_namespace"],
+                ownership["eventhub_name"],
+                ownership["consumer_group"],
+            ),
+            "RowKey": ownership["partition_id"],
+            "ownerid": ownership["owner_id"],
+        }
+        return ownership_entity
+
+    @classmethod
+    def _create_checkpoint_entity(cls, checkpoint):
+        """
+        Create a dictionary with `checkpoint` attributes.
+        """
+        checkpoint_entity = {
+            "PartitionKey": "{} {} {} Checkpoint".format(
+                checkpoint["fully_qualified_namespace"],
+                checkpoint["eventhub_name"],
+                checkpoint["consumer_group"],
+            ),
+            "RowKey": checkpoint["partition_id"],
+            "offset": checkpoint["offset"],
+            "sequencenumber": checkpoint["sequence_number"],
+        }
+        return checkpoint_entity
+
+    def _update_ownership(self, ownership, **kwargs):
+        """_update_ownership mutates the passed in ownership."""
+        try:
+            ownership_entity = TableCheckpointStore._create_ownership_entity(ownership)
+            metadata = self._table_client.update_entity(
+                mode=UpdateMode.REPLACE,
+                entity=ownership_entity,
+                etag=ownership["etag"],
+                match_condition=MatchConditions.IfNotModified,
```

Comment:
hmm, now that I think about it, why are we setting this as the MatchCondition? is this what Baffy did in JS?

Reply:
I think we should remove this, b/c it implies that we only update the entity if it has not been modified before. However, if we call this method twice (or more) on the same entity, it should update it twice  (or more).

Reply:
ok

Reply:
so we actually need it cause without the match condition it will not update the owner_id and raise an exception

Reply:
that's weird, but since we're not releasing this rn, I can check if this is the right approach later